client/src/panels/decompositionPanel.js

```diff
@@ -104,6 +105,32 @@ class DecompositionPanel extends React.Component {
     this.updateDataModel('' + result.maxPersistenceLevel);
   }
 
+  componentDidMount() {
+    if (this.state.decompositionMode == 'Morse-Smale') {
+      let k = 15; // num nearest neighbors to consider when generating M-S complex for a dataset
```

Comment:
Why not simply call this.fetchDecomposition here?
@kylinmb

Reply:
I still would like to know in case I'm missing something. But regardless, I'm going to consolidate all calls to fetch a new M-S to ensure parameters all come from a single source (and that source will be the loaded dataset if it's been pre-computed, i.e., there will be no dynamic partitioning in that case).

Reply:
I set the initial decomposition in the constructor as part of this.state. These defaults are used in the componentDidMount methods but they are also used as the default settings in the render function for the form controls. Once the component mounts using the default settings it calls the fetchMorseSMalePersistence and the updateDataModel methods, these are used to draw the embedding and ms window. I did this so the user did not have to select these items when the window first loaded but instead loaded defaults.

Reply:
Still on the todo list for code consolidation, for which I just created issue #105.